client/state/account-recovery/reducer.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+
+import {
+	ACCOUNT_RECOVERY_FETCH,
+	ACCOUNT_RECOVERY_FETCH_SUCCESS,
+	ACCOUNT_RECOVERY_FETCH_FAILED,
+} from 'state/action-types';
+
+const reducer = createReducer( {}, {
```

Comment:
I'll leave it to you to decide since I could see it going either way, but [`combineReducers`](http://redux.js.org/docs/api/combineReducers.html) may be a utility to consider to split unrelated concerns of the state, like `isFetching` from the response data. I think this subtree in particular could be an exception since the state is very much tied to the request (whereas posts, for example, could be received from other sources like editing, Reader, etc).

One particularly handy characteristic of `combineReducers` is that it manages returning the same state reference if none of the reducers returns a changed value when an action occurs. This avoids rerenders of a `connect`ed component or cachebusting of a [memoized selector](https://github.com/Automattic/wp-calypso/tree/master/client/lib/create-selector) if no change has occurred to the relevant subproperty. Here, you're always returning a new state object so even if the properties of account recovery don't change as a result of the fetch starting, anything depending on those properties will think that it has.